### PR TITLE
feat: add prettifier

### PR DESCRIPTION
**Pull request by gabyzif**
## Instructions

This PR introduces a new SVG Prettifier tool to streamline the process of uploading, prettifying, and integrating SVG icons into the project. The goal is to ensure consistency in icon styling and prevent issues with duplicate file names, making it easier for developers to maintain and update the icon set.

Why We're Doing This
To improve the process of adding icons to the project, we added a tool automatically enforces these rules (such as removing unnecessary attributes and ensuring proper dimensions), reducing the risk of inconsistencies. Additionally, the system now handles duplicate filenames gracefully, preventing accidental overwrites and ensuring clean integration of new icons.

https://github.com/user-attachments/assets/a64d8e11-5f82-45a7-8bab-24b2370dd160

- {provide a thorough description of the changes}

## Testing Plan

- [ ] Was this tested locally? If not, explain why.
- {explain how this has been tested, and what, if any, additional testing should be done}

## Reference Issue (For mParticle employees only. Ignore if you are an outside contributor)

- Closes https://go.mparticle.com/work/REPLACEME

### CODE REVIEW

**Comment by ilyusha** [comment]
### Review Forge
[Click Here for AI Insights](https://supreme-adventure-m612l2p.pages.github.io?prUrl=https://github.com/mParticle/aquarium/pull/404)

**Comment by gabyzif** [comment]
TODO: add better docs to the icon submition process

File: docs/icons.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Icons Documentation
+
+### Adding and Updating Icons
+
+The Design team provides SVG files with the final icon names. These SVG files must be minified and follow the project's styling guidelines. You can prettify them using the `SVGPrettifier` tool, which ensures the correct size and properties are applied to the SVG.
+
+#### Steps to Add a New Icon:
+
+1. **Prettify the SVG**:  
+   Use the `SVGPrettifier` script or UI to clean up the SVG file and ensure it meets the necessary formatting standards (size, properties, etc.). This feature will be available soon within the Aquarium project.
```
Comment (tibuurcio) [inline docs/icons.md:10]: > This feature will be available soon within the Aquarium project.

This PR adds it, no? We can remove that.

File: package.json [hunk 2]
```diff
@@ -43,6 +43,8 @@
     "@storybook/react-vite": "8.1.10",
     "@storybook/test": "8.1.10",
     "@storybook/test-runner": "0.18.2",
+    "@types/jsdom": "^21.1.7",
+    "@types/multer": "^1.4.12",
```
Comment (tibuurcio) [inline package.json:47]: Let's fix the versions to follow the projects standard

File: server/prettify-server.cjs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+// Use memory storage for uploaded files
+const storage = multer.memoryStorage();
+const upload = multer({ storage: storage });
+
+const server = http.createServer((req, res) => {
+  // Add CORS headers
+  res.setHeader('Access-Control-Allow-Origin', '*');  // Allow requests from any origin
+  res.setHeader('Access-Control-Allow-Methods', 'POST');  // Allow POST requests
+  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');  // Allow content-type header
```
Comment (tibuurcio) [inline server/prettify-server.cjs:19]:
```suggestion
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
```

self explanatory

File: src/utils/SvgPrettfier.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { type Meta, type StoryObj } from '@storybook/react'
+import { SVGPrettifier } from './SvgPrettifier'
+
+// SVG Prettifier Documentation
+const svgPrettifierDocumentation = `
```
Comment (tibuurcio) [inline src/utils/SvgPrettfier.stories.tsx:5]: Hopefully we will be able to write into `.mdx` files in the future 🙏🏼

**Review by tibuurcio — COMMENTED** [review]
Nice work! Please review some of the comments in the code, most of it is self-explanatory IMHO. Other than that, good to go!

**Comment by tibuurcio** [comment]
Also looks like Prettier is not running in the `./server` folder @gabyzif. Would be interesting to have it running so we can maintain the same format in the project.

File: package.json [hunk 5]
```diff
@@ -57,8 +59,10 @@
     "factory.ts": "1.4.1",
     "http-server": "14.1.1",
     "husky": "9.0.11",
-    "jsdom": "24.0.0",
+    "jsdom": "^24.1.3",
+    "multer": "^1.4.5-lts.1",
     "prettier": "3.3.2",
+    "sanitize-filename": "^1.6.3",
```
Comment (danielsiemens) [inline package.json:65]: This package hasn't gotten an update in 5 years, and it's really just a simple function. Do we need to add an extra dependency for this?

**Review by danielsiemens — APPROVED** [review]

File: src/utils/svg-prettifier/prettifier.cjs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+const { JSDOM } = require('jsdom')
+const fs = require('fs')
+const path = require('path')
+const sanitizeFilename = require('sanitize-filename')
```
Comment (danielsiemens) [inline src/utils/svg-prettifier/prettifier.cjs:4]: Oops this looks like it was leftover

**Review by danielsiemens — APPROVED** [review]

**Review by tibuurcio — APPROVED** [review]

**Pull request merged**